### DISCUSSION TITLE
Is the tool working right now?

### DISCUSSION THREAD

**Discussion by cheezedawg** [davetapley/helium-tax#86] [Q&A] [answered]
I see the change #83 a few days ago, but the helium API has been struggling recently so I wonder if there are other issues that might be preventing it from working even with the cached oracle prices. I am unable to generate anything.  Using the hotspot address, it tells me there were no rewards during 2021 (there were), and using the wallet address it just spins for a while and never gives any message or output.

**Comment by mattz608** [comment]
I'm having a similar problem. Came here to see if anyone else was experiencing the same.

**Comment by cheezedawg** [comment, marked as answer]
I see a bunch of errors in the console that probably explain my problem.  First, the call to get the timezone seems to be failing, and then all of the calls to the helium API are either rejected with 429 (rate limit) or blocked completely.  A sampling:

I edited my location and address out of these :)
 GET https://enz4qribbjb5c4.m.pipedream.net/?lat=xxxxxx&lng=xxxxxxxxx net::ERR_CONNECTION_REFUSED

GET https://api.helium.io/v1/validators/xxaddressxx 404

Access to XMLHttpRequest at 'https://api.helium.io/v1/accounts/xxaddressxx/hotspots' from origin 'https://davetapley.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

GET https://api.helium.io/v1/accounts/xxaddressxx/hotspots net::ERR_FAILED 429

**Reply by davetapley** [reply]
Thanks for the info!

Frustratingly it's working for me right now, but I'll address each error:

1. Can you try the time zone endpoint from another ISP (e.g. on cell phone)? I want to rule out Pipedream (the host I use) blocking it because they think it suspicious. You're welcome to use [Null Island](https://enz4qribbjb5c4.m.pipedream.net/?lat=0&lng=0) and you should get `Etc/GMT` back.

3. 404 on `validators/` is expected and ok if not a validator address.

5. The `No 'Access-Control-Allow-Origin' header` error I something I need to deal with, but it shouldn't stop it running.

7. The 429s are the Helium API rate limiting you, it will [automatically retry](https://github.com/helium/helium-js/blob/1a4529438d2d597241fb3b8dc9db2a09888fe163/packages/http/src/Client.ts#L72) up to 5 times, so if you see five 429s in a row for the same URL then it will just give up. Here's an example from mine just now of it taking trying three times: <img width="713" alt="helium-tax" src="https://user-images.githubusercontent.com/48232/163299572-3a6ea37e-24b5-4735-b46e-75ba68e9e26b.png">


That's all to say: Aside from the Pipedream / time zone issue I'm not sure what else I can do 😑

**Reply by cheezedawg** [reply]
That was actually quite helpful. I had some time tonight so I started looking into the pipedream connection. I discovered 2 things:
- My McAfee firewall software was blocking the site, so I added it to the whitelist
- The connection was still getting refused, and so I added to my Eero firewall whitelist, and...

Success! After doing this, I was able to download my CSV!  Thank you!
